Rename `BACKEND` to be `RPC_BACKEND` to be seperated from `COMMUNICATION_BACKEND` like gloo,nccl, in `rpc_test.py`

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27792 Rename `BACKEND` to be `RPC_BACKEND` to be seperated from `COMMUNICATION_BACKEND` like gloo,nccl, in `rpc_test.py`**

Close https://github.com/pytorch/pytorch/issues/27232

Differential Revision: [D5474297](https://our.internmc.facebook.com/intern/diff/D5474297/)

xush6528 committed:
Rename `BACKEND` to be `RPC_BACKEND` to be seperated from `COMMUNICATION_BACKEND` like gloo,nccl, in `rpc_test.py`

Close https://github.com/pytorch/pytorch/issues/27232

Differential Revision: [D5474297](https://our.internmc.facebook.com/intern/diff/D5474297/)

[ghstack-poisoned]

xush6528 committed:
Update on "Rename `BACKEND` to be `RPC_BACKEND` to be seperated from `COMMUNICATION_BACKEND` like gloo,nccl, in `rpc_test.py`"


Close https://github.com/pytorch/pytorch/issues/27232

Differential Revision: [D5474297](https://our.internmc.facebook.com/intern/diff/D5474297/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
Looks good, thanks!

xush6528 committed:
Update on "Rename `BACKEND` to be `RPC_BACKEND` to be seperated from `COMMUNICATION_BACKEND` like gloo,nccl, in `rpc_test.py`"


Close https://github.com/pytorch/pytorch/issues/27232

Differential Revision: [D5474297](https://our.internmc.facebook.com/intern/diff/D5474297/)

[ghstack-poisoned]
